Lib/test/test_socket.py

```diff
@@ -44,6 +46,22 @@
 except ImportError:
     _socket = None
 
+def get_cid():
+    if fcntl is None:
+        return None
+    if not os.path.exists("/dev/vsock"):
```

Comment:
Please rewrite this block with proper context manager. You can skip the os.path.exists check, too.

```
try:
    with open(...) as f:
        r = fcntl.ioctl(f, socket.IOCTL_VM_SOCKETS_GET_LOCAL_CID, "    ")
except OSError:
    return None
else:
    return struct.unpack("I", r)[0]
```